code/game/objects/items/toys.dm

```diff
@@ -547,70 +600,510 @@
+	..()
+
+/obj/item/toy/prize/attack(mob/living/carbon/human/target, mob/living/carbon/human/user) //src is user's toy
+	if(target == user)
+		to_chat(user, "<span class='notice'>Target another toy mech if you want to start a battle with yourself.</span>")
+		return
+	else if(user.a_intent != INTENT_HARM)
+		if(wants_to_battle) //prevent spamming someone with offers
+			to_chat(user, "<span class='notice'>You already are offering battle to someone!</span>")
+			return
+		if(!check_battle_start(user)) //if the user's mech isn't ready, don't bother checking
+			return
+
+		for(var/obj/item/I in target.held_items)
+			if(istype(I, /obj/item/toy/prize)) //if you attack someone with a mech who's also holding a mech, offer to battle them
+				var/obj/item/toy/prize/P = I
+				if(!P.check_battle_start(target, null, user)) //check if the attacker mech is ready
+					break
+
+				//slap them with the metaphorical white glove
+				if(P.wants_to_battle) //if the target mech wants to battle, initiate the battle from their POV
+					mecha_brawl(P, target, user) //P = challenger's mech / SRC = defender's mech / user = defender / target = challenger
+					P.wants_to_battle = FALSE
+					return
+
+		//extend the offer of battle to the other mech
+		to_chat(user, "<span class='notice'>You offer battle to [target.name]!</span>")
+		to_chat(target, "<span class='notice'><b>[user.name] wants to battle with [user.p_their()] [name]!</b> <i>Attack them with a toy mech to initiate combat.</i></span>")
+		wants_to_battle = TRUE
+		addtimer(CALLBACK(src, .proc/withdraw_offer, user), 6 SECONDS)
+		return
+
+	..()
+
+/**
+  * Resets the request for battle.
+  *
+  * For use in a timer, this proc resets the wants_to_battle variable after a short period.
+  * Arguments
+  *		user: the user wanting to do battle
+  */
+/obj/item/toy/prize/proc/withdraw_offer(mob/living/carbon/user)
+	if(wants_to_battle)
+		wants_to_battle = FALSE
+		to_chat(user, "<span class='notice'>You get the feeling they don't want to battle.</span>")
+
+/obj/item/toy/prize/suicide_act(mob/living/carbon/user)
+	if(in_combat)
+		to_chat(user, "<span class='notice'>[src] is in battle, let it finish first.</span>")
+		return
+	
+	user.visible_message("<span class='suicide'>[user] begins a fight [user.p_they()] can't win with [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	
+	in_combat = TRUE
+	sleep(1.5 SECONDS)
+	SpinAnimation(5, 0)
+	user.adjustBruteLoss(20)
+	user.adjustStaminaLoss(40)
+	sleep(1 SECONDS)
+	user.SpinAnimation(5, 0)
+	combat_health-- //we scratched it!
+	sleep(1 SECONDS)
+	SpinAnimation(5, 0)
+	user.adjustBruteLoss(15)
+	user.adjustStaminaLoss(30)
+	sleep(1 SECONDS)
+	SpinAnimation(5, 0)
+	user.adjustBruteLoss(15)
+	user.adjustStaminaLoss(30)
+	sleep(1 SECONDS)
+	if(!quiet)
+		say(special_attack_cry + "!!")
+		user.emote("scream")
+	else
+		SpinAnimation(5, 0)
+	sleep(1.5 SECONDS)
+	if(!in_range(src, user)) //run away? coward
+		say("PATHETIC")
```

Comment:
What if the user runs away or is already dead before this point? Unless I'm missing something, it'll probably cause a runtime.

Maybe create a custom sleep proc for mechs that checks if theyre within range of their target, if their target exists, if the target is alive, if the attacker exists and is alive, etc., and returns a boolean accordingly?

Reply:
i don't think it runtimes, but it was messy, so i reworked it + implemented a "sleep + check" proc, good suggestion